fate/src/main/java/org/apache/accumulo/fate/util/Retry.java

```diff
@@ -0,0 +1,356 @@
+     */
+    RetryFactory createFactory();
+
+    /**
+     * Create a single Retry object with the currently configured builder settings.
+     *
+     * @return a retry object from this builder's settings
+     */
+    Retry createRetry();
+  }
+
+  public interface RetryFactory {
+    /**
+     * Create a single Retry object from this factory's settings.
+     *
+     * @return a retry object from this factory's settings
+     */
+    Retry createRetry();
+  }
+
+  public static NeedsRetries builder() {
+    return new RetryFactoryBuilder();
+  }
+
+  private static class RetryFactoryBuilder implements NeedsRetries, NeedsRetryDelay, NeedsTimeIncrement, NeedsMaxWait, NeedsLogInterval, BuilderDone,
+      RetryFactory {
+
+    private boolean modifiable = true;
+    private long maxRetries;
+    private long initialWait;
+    private long maxWait;
+    private long waitIncrement;
+    private long logInterval;
+
+    RetryFactoryBuilder() {}
+
+    private void checkState() {
+      Preconditions.checkState(modifiable, "Cannot modify this builder once 'createFactory()' has been called");
+    }
+
+    @Override
+    public NeedsRetryDelay infiniteRetries() {
+      checkState();
+      this.maxRetries = -1;
+      return this;
+    }
+
+    @Override
+    public NeedsRetryDelay maxRetries(long max) {
+      checkState();
+      Preconditions.checkArgument(max >= 0, "Maximum number of retries must not be negative");
+      this.maxRetries = max;
+      return this;
+    }
+
+    @Override
+    public NeedsTimeIncrement retryAfter(long duration, TimeUnit unit) {
+      checkState();
+      this.initialWait = unit.toMillis(duration);
+      Preconditions.checkArgument(initialWait >= 0, "Initial waiting period must not be negative");
+      return this;
+    }
+
+    @Override
+    public NeedsMaxWait incrementBy(long duration, TimeUnit unit) {
+      checkState();
+      this.waitIncrement = unit.toMillis(duration);
+      Preconditions.checkArgument(waitIncrement >= 0, "Amount of time to increment the wait between each retry must not be negative");
+      return this;
+    }
+
+    @Override
+    public NeedsLogInterval maxWait(long duration, TimeUnit unit) {
+      checkState();
+      this.maxWait = unit.toMillis(duration);
+      Preconditions.checkArgument(maxWait >= initialWait, "Maximum wait between retries must not be less than the initial delay");
+      return this;
+    }
+
+    @Override
+    public BuilderDone logInterval(long duration, TimeUnit unit) {
+      checkState();
+      this.logInterval = unit.toMillis(duration);
+      Preconditions.checkArgument(logInterval >= 0, "The amount of time between logging retries must not be negative");
+      return this;
+    }
+
+    @Override
+    public RetryFactory createFactory() {
+      this.modifiable = false;
+      return this;
+    }
+
+    @Override
+    public Retry createRetry() {
+      return new Retry(maxRetries, initialWait, waitIncrement, maxWait, logInterval);
```

Comment:
I manually checked the args and they line up.  Then I looked for a test that does this, it seems like the `properArgumentsInRetry` test should check that these line up.

Reply:
It does.